packages/server/routes/api/events.js

```diff
@@ -280,4 +281,47 @@ router.get(
   }
 );
 
+router.get(
+  "/:guildId/upcoming",
+  AuthController.authenticate,
+  guildIdValidator,
+  async (request, response) => {
+    const result = validationResult(request);
+
+    // if validation result is not empty, errors occurred
+    if (!result.isEmpty()) {
+      response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+      return;
+    }
+
+    try {
+      const currDate = new Date();
+
+      const validatedData = matchedData(request);
+      const guildId = validatedData.guildId;
+
+      //Should return a list of events from the guild that are upcoming and in ascending order
+      const upcoming = await EventController.getUpcomingEvents(
+        currDate,
+        guildId
+      );
+
+      response.status(200).json({
+        status: "success",
+        data: {
+          upcoming: upcoming,
+        },
```

Comment:
Try to be more specific with naming response data so it's more clear to the frontend.
```js
{
  status: "success",
  data: {
    upcomingEvents: upcoming
  }
}
```